### ISSUE TITLE
Boundary of detected objects is not visible or detection is failing.

### ISSUE THREAD

**Issue by Kaushal28** [tensorflow/models#2773]
I've just copy pasted the code and ran it. It runs perfectly fine, but at last, when result of testing is shown, the original image is shown without labels and boundary of detected objects. I've also asked the same on stack overflow. You can find it here: https://stackoverflow.com/questions/47237388/nothing-is-being-detected-tensorflow-object-detection-api

Please help.

**Comment by johannesschaefer101** [comment]
I have got the same problem. No error messages.

**Comment by andresferreira** [comment]
I also have the same problem

**Comment by Georg-W** [comment]
Same for me.
Using an older version ('ssd_mobilenet_v1_coco_11_06_2017') worked.

**Comment by johannesschaefer101** [comment]
also works with "ssd_mobilenet_v1_coco_11_06_2017"

**Comment by jch1** [comment]
Can you all say which version of tensorflow you are running?  

e.g., with print tf.__version__

**Comment by johannesschaefer101** [comment]
version 1.4

**Comment by jch1** [comment]
hmmm strange :)  I thought I had tested this.  Can you check to see if the Faster R-CNN models work alright?

And can you verify that it is 1.4.0 and not 1.4.0rc1?

**Comment by johannesschaefer101** [comment]
@jch1 

- I use version **1.4.0**
- @andresferreira workaround **works** for me. but the faster R-CNN version is multiple times slower than the ssd modul

**Comment by mghildiy** [comment]
I am also facing this issue. And if model has this issue,does it mean training with that model would also be not a good idea?Actually I started the training with my dataset and its stuck at:
global_step/sec: 0
Its stuck for almost 9 hours.
I am training on CPU.

**Comment by jch1** [comment]
Also, just to collect a bit more info, can you say what OS you are using please?

**Comment by mghildiy** [comment]
I am using Windows 10.

**Comment by Kaushal28** [comment]
I'm using windows 8.1, with tensor flow version 1.4

**Comment by mghildiy** [comment]
I have a question about the nature of data one should train with Tensorflow object detection models.
My data set is satellite imagery, and my aim is to detect vehicles. Are TF object detection models good fit for this use case? When I try to train with 'ssd_mobilenet_v1_coco_11_06_2017', python crashes(3.6.3) after 2 global steps.

**Comment by hiewliwen** [comment]
Likewise.
Win 10 Pro
TF 1.4

ssd_mobilenet_v1_coco_2017_11_08 doesn't work.
ssd_mobilenet_v1_coco_11_06_2017 works.

**Comment by sandromario** [comment]
Same problem here, on Linux (Ubuntu) this time. The net named "ssd_mobilenet_v1_coco_2017_11_08" doesn't work; it produces abnormally low scores.

**Comment by kaka-lin** [comment]
Same problem, 
use ssd_mobilenet_v1_coco_11_06_2017 works. -> run function draw_bounding_box_on_image
use ssd_mobilenet_v1_coco_2017_11_08 doesn't work. -> No run function draw_bounding_box_on_image

**Comment by Kaushal28** [comment]
#2788

**Comment by Kaushal28** [comment]
#2788

**Comment by Kaushal28** [comment]
If anyone want to use old  `ssd_mobilenet_v1_coco`, then you can find it [here].(http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v1_coco_11_06_2017.tar.gz) It is working and perfect for applications where you need faster detection instead of higher accuracy.

**Comment by tombstone** [comment]
@Kaushal28  Could you please clarify two things.
1. Are you using the latest version of object detection code ? (If not can you try this ?. The new models in the zoo need the latest object detection code to work correctly)
2. Are you on python 3 ?

**Comment by tombstone** [comment]
I am able to reproduce this with python 2.7 , tf 1.4 and latest object detection code. Taking a look at his.

**Comment by tombstone** [comment]
Please try the updated ssd models from the detection zoo again. That should fix the problem.

**Comment by Pawan2721** [comment]
Hold onnn!!!!!! I got the proper output for 'ssd_mobilenet_v1_coco_11_06_2017' in Ubuntu 16.04 using tensorflow 1.4.  I tried the same on Windows bt I dint get!!!!

**Comment by Kaushal28** [comment]
Why this issue is closed even if it is not solved?

**Comment by Kaushal28** [comment]
@tombstone yes I'm using latest version of code with Python 3.x

**Comment by tombstone** [comment]
@kaushal28 this was closed automatically as it was referenced in the PR that updated the model zoo. This problem should go away if you use the latest model from the zoo on tensorflow 1.4 . Here's a link for your convenience  http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v1_coco_2017_11_17.tar.gz. . please feel free to reopen if this doesn't work for you

**Comment by tombstone** [comment]
@Pawan2721 please verify that you have tensorflow 1.4 installed on Windows and you are using the latest object detection code and model http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v1_coco_2017_11_17.tar.gz. If you still continues to have problems on Windows let's open a separate issue for it.

**Comment by Jreyno40** [comment]
I'm adding a comment here; I'm seeing the same issue of no bounding boxes showing up, but when training models using the sample configurations and not in the entry ipynb.

**Comment by mghildiy** [comment]
I tried training with this new model, but training fails after few steps.
Here is the SO link:
https://stackoverflow.com/questions/47484883/tensorflow-object-detection-training-makes-python-crash

**Comment by blaster666** [comment]
Now it can visible with tensorflow 1.4.1

**Comment by alessandrofrancesconi** [comment]
I'm experiencing a similar problem on TF 1.4.0, by training the faster_rcnn_inception_v2 model but from scratch (so I'm not working on a pre-trained model).
I've already posted a question on StackOverflow: https://stackoverflow.com/questions/48227959/object-detection-boxes-are-lost-at-second-evaluation-step

Maybe is something related, maybe not... Is there someone who can help me? Thank you.

**Comment by marcoromelli** [comment]
@alessandrofrancesconi I'm facing a similar issue. Did you solve the problem?

**Comment by Pawan2721** [comment]
Which model have u used!!!?

**Comment by marcoromelli** [comment]
@Pawan2721 I tried both `ssd_mobilenet_v1_coco` and `ssd_inception_v2_coco` from the model zoo.
I'm using TF 1.6.0 cpu-only on MacOS.

**Comment by Pawan2721** [comment]
Use ths version of mobilnet model
ssd_mobilenet_v1_coco_11_06_2017

**Comment by TerenceCYJ** [comment]
Same Problem！
I am using TF1.4.0 on Ubuntu 16.04
I guess the frozen inference graphs are generated using the different release version of Tensorflow.
Frozen inference graph can be regenerated using your current version of Tensorflow by re-running the
[exporter](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/exporting_models.md)
**# It works!** @Kaushal28

**Comment by devharsh** [comment]
same issue.. no objects detected..

i used this model for transfer learning: ssdlite_mobilenet_v2_coco_2018_05_09

os: windows 7
python: Python 3.5.2 |Anaconda custom (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)] on win32

**Comment by CHANDL3** [comment]
I have the same problem when using ssdlite_mobilenet_v2_coco_2018_05_09 model on windows 7.

**Comment by CrazyHrodgar** [comment]
Appartently there is problem with the models used. Use this line `MODEL_NAME = 'faster_rcnn_resnet101_coco_2017_11_08'` in the section "Model preparation".
Hope this will help.

os: windows 10
python:  python 3.6.8 | Anaconda 4.6.7
tensorflow gpu 1.12

**Comment by soufianesabiri** [comment]
I got 0 score in all detections... Please help!

**Comment by Shaul-Z** [comment]
I have recently installed TensorFlow 1.14 (cpu & gpu) using Anaconda2 on Windows 10. I followed Sentdex tutorial and everything went fine: at the end it detected the dog and the people on the beach. I even tried another dog picture from google and it detected it just fine. 
Problem is that after I finished training my own model it didn’t detect anything. No boxes nor boundaries at all. I used more than 100 pics for training.

**Comment by CarrierDiagnostics** [comment]
Did you get a loss for each step or was it always 0?

**Comment by Shaul-Z** [comment]
Yes, I got loss for each step.

**Comment by CarrierDiagnostics** [comment]
> Yes, I got loss for each step.

I got it to work using 
https://github.com/EdjeElectronics/TensorFlow-Object-Detection-API-Tutorial-Train-Multiple-Objects-Windows-10

SSD still doesn't work I had to switch to:

faster_rcnn_inception_v2_coco_2018_01_28

But it now works great. Get at least 15k steps.

**Comment by Nazeef-Ul-Haq** [comment]
I trained my model using SSD_inception_v2 and facing same issue. can anyone guide how to fix this?

**Comment by rybradlax** [comment]
> I trained my model using SSD_inception_v2 and facing same issue. can anyone guide how to fix this?
Nazeef-Ul-Haq Tensorflow is really buggy even on stable versions, so as I said before it could be the order of the y and x mins and maxes in your tfrecords, but also the only version of Tensorflow that Tensorflow can assure will work most of the time with most models is 1.12.0, and I'd try using the ssd mobilenet v1 from any month in 2017.

**Comment by rebeen** [comment]
could you please help me to evaluate my SSD model since I used this https://github.com/qfgaohao/pytorch-ssd
with coco dataset and v-coco annotation now I do not know how to get the result.json file because I have got model.pth file

**Comment by gauravkantrod** [comment]
I have trained my custom model on tensor-flow 2.5 and ssd_resnet101_v1_fpn_640_640_coco17 and my tensor-flow model detects nothing while detecting.
Note --  No error while training the model.

**Comment by Annieliaquat** [comment]
> I've just copy pasted the code and ran it. It runs perfectly fine, but at last, when result of testing is shown, the original image is shown without labels and boundary of detected objects. I've also asked the same on stack overflow. You can find it here: https://stackoverflow.com/questions/47237388/nothing-is-being-detected-tensorflow-object-detection-api
> 
> Please help.

Can you please tell me how did you solved the problem.. My model is training and testing perfectly but the font size is too small that not even visible. what can I do to solve the problem

**Comment by nkulkarni3297** [comment]
I am training my custom dataset for **Sign Language to Text Conversion** on Tensorflow v2.10. I am using the checkpoints from ssd_mobilenet_v2_fpnlite_320x320_coco17_tpu-8 on 36 classes. Initially it didn't worked with any class so I stepped back and started with 10 classes and slowly adding the classes. Till 25 classes everything was working fine. When I added 26th class, it started with the reduction of class detection. So, I trained 26th class with the last checkpoint from class 25th and it almost worked. Similarly I started moving ahead. Till 31 classes it was detecting the objects nicely. But when I added class 32nd, it is not giving accurate results. Is there any possible solutions for it including the fine tuning of the model? 

Also, right now with 31 classes working, it still has some issues with the colors. For instance, my training and testing set contains images from same day so I have a specific color of dress repeatedly in the dataset. So on testing the detections on webcam, model randomly shows the detection on anything in that specific color without the actual object on webcam. So I am confused on why that is happening too?
